### PR TITLE
adding created by and created at to some models

### PR DESCRIPTION
**Pull request by bochka123**

### CODE REVIEW

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Context/EntityConfigurations/UserConfig.cs [hunk 1]
```diff
@@ -26,6 +26,9 @@ public void Configure(EntityTypeBuilder<User> builder)
 
             builder.HasMany(u => u.Lessons)
                 .WithMany(l => l.Subscribers);
+
+            builder.Property(u => u.CreatedAt)
+                .HasDefaultValueSql("getdate()");
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Context/EntityConfigurations/UserConfig.cs:31]: Have you checked if it returns generic date? Not connected to timezone?

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Common/ICreatedBy.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EasySpeak.Core.DAL.Entities
+{
+    public interface ICreatedBy
+    {
+        public string CreatedBy { get; set; }
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Common/ICreatedBy.cs:11]: Remove public

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Call.cs [hunk 3]
```diff
@@ -6,11 +6,12 @@
...
 namespace EasySpeak.Core.DAL.Entities
 {
-    public class Call : Entity<long>
+    public class Call : Entity<long>, ICreatedBy
     {
         public long ChatId { get; set; }
         public Chat Chat { get; set; } = null!;
         public DateTime StartedAt { get; set; }
         public DateTime? FinishedAt { get; set; }
+        public string CreatedBy { get; set; } = string.Empty;
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Call.cs:15]: Can't we set it to the same type as we have User.Id type? Why string?

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Call.cs [hunk 4]
```diff
@@ -6,11 +6,12 @@
 
 namespace EasySpeak.Core.DAL.Entities
 {
-    public class Call : Entity<long>
+    public class Call : Entity<long>, ICreatedBy
```
Comment (potapchuk-anna) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Call.cs:9]: Maybe it is better to create class, that extands Entity and has field CreatedBy? See the example https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-easyspeak/blob/80f430b33f73580294a08f4949ca759910dc7aa7/backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Common/AuditEntity/AuditEntity.cs#L3

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Context/EasySpeakCoreContext.cs [hunk 5]
```diff
@@ -18,14 +22,40 @@ public class EasySpeakCoreContext : DbContext
...
+        {
+            var entities = ChangeTracker.Entries().Where(x => x.Entity is Entity<long> && (x.State == EntityState.Added));
+
+#pragma warning disable CS8602 // Dereference of a possibly null reference.
+            var userId = _httpContextAccessor.HttpContext.User.FindFirst(ClaimTypes.Email).Value;
+#pragma warning restore CS8602 // Dereference of a possibly null reference.
+
+            var currentUsername = !string.IsNullOrEmpty(userId)
+                ? userId
+                : throw new ArgumentNullException("User was not found");
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Context/EasySpeakCoreContext.cs]: Won't it be checked automatically by foreign key?

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Context/EasySpeakCoreContext.cs [hunk 6]
```diff
@@ -18,14 +22,40 @@ public class EasySpeakCoreContext : DbContext
...
+
+            foreach (var entity in entities)
+            {
+                if (entity.State == EntityState.Added)
+                {
+                    ((ICreatedBy)entity.Entity).CreatedBy = currentUsername;
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Context/EasySpeakCoreContext.cs:56]: Let's make references as Ids, not usernames

**Review by Anatolii-Ivanov — COMMENTED** [review]
Can you also fix "Unchanged files with check annotations" section?

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Question.cs [hunk 7]
```diff
@@ -6,14 +6,13 @@
 
 namespace EasySpeak.Core.DAL.Entities
 {
-    public class Question : Entity<long>
+    public class Question : AuditEntity<long>
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Question.cs:9]: CreatedBy not needed here, as it's always connected to a lesson

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Subquestion.cs [hunk 8]
```diff
@@ -6,7 +6,7 @@
 
 namespace EasySpeak.Core.DAL.Entities
 {
-    public class Subquestion : Entity<long>
+    public class Subquestion : AuditEntity<long>
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Entities/Subquestion.cs:9]: The same

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by potapchuk-anna — APPROVED** [review]

**Pull request merged**